components/hab/src/main.rs

```diff
@@ -499,8 +498,8 @@ fn sub_pkg_export(ui: &mut UI, m: &ArgMatches<'_>) -> Result<()> {
     let url = bldr_url_from_matches(&m)?;
     let channel = m
         .value_of("CHANNEL")
-        .and_then(|c| Some(c.to_string()))
-        .unwrap_or(channel::default());
+        .map(str::to_string)
+        .unwrap_or_else(channel::default);
```

Comment:
Note that we *don't* want `unwrap_or_default` here, as `String::default()` != `channel::default()`

Reply:
Ugh... my kingdom for a proper `Channel` type!

https://github.com/habitat-sh/core/blob/22f09fc7ca4e8dd87a655cd2186eff1c20d6eea9/components/core/src/channel.rs